File: ios/XiriGoEcommerceTests/Feature/Onboarding/Fakes/FakeOnboardingRepository.swift
import Foundation
@testable import XiriGoEcommerce

// MARK: - FakeOnboardingRepository

/// In-memory fake implementation of `OnboardingRepository` for use in unit tests.
/// Avoids touching UserDefaults — each test instance starts with a clean state.
final class FakeOnboardingRepository: OnboardingRepository, @unchecked Sendable {
    // MARK: - Configurable State

    var hasSeen: Bool = false
    var setOnboardingSeenCallCount: Int = 0

    // MARK: - OnboardingRepository

    func hasSeenOnboarding() async -> Bool {
        hasSeen
    }

    func setOnboardingSeen() async {
        hasSeen = true
        setOnboardingSeenCallCount += 1
    }
}

File: android/app/src/test/java/com/xirigo/ecommerce/feature/onboarding/repository/FakeOnboardingRepository.kt
package com.xirigo.ecommerce.feature.onboarding.repository

import com.xirigo.ecommerce.feature.onboarding.domain.repository.OnboardingRepository

class FakeOnboardingRepository : OnboardingRepository {

    private var hasSeen: Boolean = false
    var setOnboardingSeenCallCount: Int = 0

    fun setHasSeen(value: Boolean) {
        hasSeen = value
    }

    override suspend fun hasSeenOnboarding(): Boolean = hasSeen

    override suspend fun setOnboardingSeen() {
        hasSeen = true
        setOnboardingSeenCallCount++
    }
}
